Spigot-Server-Patches/0315-Implement-Expanded-ArmorStand-API.patch

```diff
@@ -0,0 +1,144 @@
+From 7b83f4e16ad55d668d9888df9958a25dd373a4ed Mon Sep 17 00:00:00 2001
+From: willies952002 <admin@domnian.com>
+Date: Thu, 26 Jul 2018 02:25:46 -0400
+Subject: [PATCH] Implement Expanded ArmorStand API
+
+Add the following:
+- Add proper methods for getting and setting items in both hands. Deprecates old methods
+- Enable/Disable slot interactions
+
+diff --git a/src/main/java/net/minecraft/server/EntityArmorStand.java b/src/main/java/net/minecraft/server/EntityArmorStand.java
+index e9a746f13..3b2b94d8c 100644
+--- a/src/main/java/net/minecraft/server/EntityArmorStand.java
++++ b/src/main/java/net/minecraft/server/EntityArmorStand.java
+@@ -36,7 +36,7 @@ public class EntityArmorStand extends EntityLiving {
+     private final NonNullList<ItemStack> bF;
+     private boolean bG;
+     public long h;
+-    private int bH;
++    private int bH;public void setDisabledSlots(int i) { bH = i;} public int getDisabledSlots() { return bH ;} // Paper - OBFHELPER
+     private boolean bI;
+     public Vector3f headPose;
+     public Vector3f bodyPose;
+@@ -362,6 +362,7 @@ public class EntityArmorStand extends EntityLiving {
+         return enumitemslot;
+     }
+ 
++    public boolean isSlotDisabled(EnumItemSlot slot) { return this.c(slot); } // Paper - OBFHELPER
+     private boolean c(EnumItemSlot enumitemslot) {
+         return (this.bH & 1 << enumitemslot.c()) != 0;
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java
+index 124c3185b..5d05b4cf7 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftArmorStand.java
+@@ -30,11 +30,13 @@ public class CraftArmorStand extends CraftLivingEntity implements ArmorStand {
+     }
+ 
+     @Override
++    @Deprecated // Paper
+     public ItemStack getItemInHand() {
+         return getEquipment().getItemInHand();
+     }
+ 
+     @Override
++    @Deprecated // Paper
+     public void setItemInHand(ItemStack item) {
+         getEquipment().setItemInHand(item);
+     }
+@@ -222,5 +224,92 @@ public class CraftArmorStand extends CraftLivingEntity implements ArmorStand {
+     public void setCanMove(boolean move) {
+         getHandle().canMove = move;
+     }
++
++    @Override
++    public ItemStack getItemInMainHand() {
++        return getEquipment().getItemInMainHand();
++    }
++
++    @Override
++    public void setItemInMainHand(ItemStack item) {
++        getEquipment().setItemInMainHand(item);
++    }
++
++    @Override
++    public ItemStack getItemInOffHand() {
++        return getEquipment().getItemInOffHand();
++    }
++
++    @Override
++    public void setItemInOffHand(ItemStack item) {
++        getEquipment().setItemInOffHand(item);
++    }
++
++    @Override
++    public ItemStack getItemInHand(org.bukkit.inventory.EquipmentSlot slot) {
```

Comment:
Actually, we should very well make this setItem/getItem(slot) and make all the slots work.